Source/Core/DolphinQt2/Translator.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <clocale>
+#include <libintl.h>
+#include <QString>
+#include <string>
+
+#include "Common/FileUtil.h"
+#include "Core/ConfigManager.h"
+#include "DolphinQt2/Translator.h"
+
+void Translator::Init()
+{
+	std::string domain = File::GetSysDirectory() + "../../locale";
+	bindtextdomain("dolphin-emu", domain.c_str());
+	textdomain("dolphin-emu");
+	setlocale(LC_ALL, "");
+
+	Language language_id = static_cast<Language>(SConfig::GetInstance().m_InterfaceLanguage);
+	if (language_id != Language::DEFAULT)
+		SetLanguage(language_id);
+
+	RegisterStringTranslator([] (const char* string) -> std::string {return gettext(string);});
+}
+
+void Translator::SetLanguage(Language language_id)
+{
+	setlocale(LC_ALL, GetLocale(language_id));
+}
+
+QString Translator::translate(const char* context, const char* source_text, const char*  disambiguation, int n) const
+{
+	return QString::fromUtf8(gettext(source_text));
+}
+
+const char* Translator::GetLocale(Language language_id)
+{
+	switch (language_id)
+	{
+	case Language::CATALAN: return "ca_ES.UTF-8";
```

Comment:
These should be in sorted order following the same layout as the enum, same with in DisplayLanguage.